libraries/helpers.rb

```diff
@@ -0,0 +1,60 @@
+module OslRepos
+  module Cookbook
+    module Helpers
+      # Define variables to use in repo urls
+      def centos_url
+        # CentOS 7 splits up ppc64 and aarch64 into a secondary architecture repo.
+        if node['platform_version'].to_i < 8
+          case node['kernel']['machine']
+          when 'aarch64', 'ppc64le', 's390x'
+            url = 'https://centos-altarch.osuosl.org'
+          else
+            url = 'https://centos.osuosl.org'
+          end
+        else
+          url = 'https://centos.osuosl.org'
+        end
+        url
+      end
+
+      # power8/9 replaces $basearch in power9 repositories
```

Comment:
Unfortunately this is incorrect and I apologize if I haven't explained this well. Here is how the logic should be:
1. If on CentOS 7 and power9? the `base_arch` should be `power9`
2. Otherwise assume `$basearch` for all other conditions

Basically, POWER9 on CentOS 7 requires a special kernel but POWER8 does not need it.